Microsoft.Toolkit.Uwp.UI.Media/Effects/Abstract/ImageEffectBase.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Toolkit.Uwp.UI.Media.Effects.Interfaces;
+
+namespace Microsoft.Toolkit.Uwp.UI.Media.Effects.Abstract
+{
+    /// <summary>
+    /// An image based effect that loads an image at the specified location
+    /// </summary>
+    public abstract class ImageEffectBase : IPipelineEffect
```

Comment:
Can some of the existing brushes in the toolkit be re-written to use these new bases? i.e. could we add the interface, and then maybe deprecate them if they're no longer needed?  Not sure if that makes sense, but if we're adding more flexibility here, just want to make sure we're not confusing folks with all the options.

Reply:
I don't think that's possible, unfortunately 😕

Brushes and effects are two completely separate concepts here. Effects are mostly just declarations for Win2D/Composition effects to then instantiate in a pipeline, when building a brush. But they have no knowledge at all of the actual brush they'll be used for. Also, the way they're structured is with a purely functional approach, so that they can be reused across different pipelines as well (just like entire pipelines are immutable and reusable too).

Because of this, they're mostly static and can't be modified or animated directly. As in, you can't target an effect from code behind and change its value - it won't be reflected in real time in the resulting brush, as the effect is not being used directly, it's just a declaration.

The only way to animate brushes or individual effects is through code-behind (and I'm working on some additional/better APIs for that). So I think at least for now we should keep those other existing brushes too, they just have a different purpose (eg. they can be modified/animated directly from XAML).